File: backend/backend/epreuve-integree/src/main/java/be/icc/epreuve/integree/models/Bid.java
package be.icc.epreuve.integree.models;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.util.Date;


@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity(name = "bids")
public class Bid {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id",nullable=false)
    private User bidder;


    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "art_id",nullable=false)
    private Art art;

    @Column(name = "bidding_price")
    private BigDecimal amount;

    @Column(name = "auction_date")
    @CreationTimestamp
    private Date bidDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "state")
    private BidState state;
}

File: art_auction_frontend/src/app/_model/bid.ts
import { Art } from "./art";
import { BidState } from "./bidState";

import { User } from "./user";

export class Bid {
    id?: number;
    bidder: User=new User();
    art: Art=new Art();
    amount: number=0;
    bidDate: Date=new Date();
    state!: BidState;
    isHighestBidder?: boolean ;
    isAuctionEnded?: boolean;
} 
